src/Codeception/Module/Symfony2.php

```diff
@@ -142,6 +143,13 @@ protected function bootKernel()
         }
         $this->kernel = new $this->kernelClass($this->config['environment'], $this->config['debug']);
         $this->kernel->boot();
+        if ($this->config['cache_router'] === true) {
+            if (null === $this->permanentServices['router']) {
```

Comment:
Nop, If router has not been cached previously, then get router from container (from the current kernel) and store instance at ````$this->permanentServices````.

I think it's clear, but I can swap the conditional body to not compare against null, and improve readability.